### PR TITLE
Update Test_TC_TSUIC_1_1.yaml - add optional check for attribute 0x2

### LINKED ISSUE #27205: [TSUIC] 1.1 Test not checking optionality
### Reproduction steps

- Run case with application that does not have attribute 0x2 enabled 
- check on global attribute list attribute fails as it's checks 0x2 unconditionally.

### Bug prevalence

Always

### GitHub hash of the SDK that was being used

v1.1

### Platform

other

### Platform Version(s)

_No response_

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by tima-q**
Basic check on attribute-list did not account for the optionality of attribute 0x2. 
Updated script to cover this based on PICS TSUIC.S.A0002

Test-plan also needs attention to update the comment there - still talks about a attribute 0x3 while that is not in spec.
Could be this is as expected as tests are marked 'removed' in the plan

Closes #27205

### CODE REVIEW

**Review by bzbarsky-apple — CHANGES_REQUESTED** [review]
Please get the test plan updated first.  This YAML must match the test plan.

**Comment by tima-q** [comment]
> Please get the test plan updated first. This YAML must match the test plan.

Understood - https://github.com/CHIP-Specifications/chip-test-plans/issues/2841 created to track test plan changes.

**Comment by woody-apple** [comment]
@tima-q Any update on next steps here?

**Comment by tima-q** [comment]
> @tima-q Any update on next steps here?

Test plan issue has been resolved - this change can be made I expect.
@bzbarsky-apple do you agree ?

File: src/app/tests/suites/certification/Test_TC_TSUIC_1_1.yaml [hunk 1]
```diff
@@ -53,7 +53,18 @@ tests:
...
+     - label:
+          "Read the optional attribute(ScheduleProgrammingVisibility) in
+          AttributeList"
+      PICS: TSUIC.S.A0002
+      command: "readAttribute"
+      attribute: "AttributeList"
+      response:
+          constraints:
+              type: list
+              contains: [2]
```
Comment (bzbarsky-apple) [inline src/app/tests/suites/certification/Test_TC_TSUIC_1_1.yaml:67]: This implements one side of the "if and only if" from the test plan.  Also needs a test that the attribute is not there if !TSUIC.S.A0002

Comment (tima-q) [reply src/app/tests/suites/certification/Test_TC_TSUIC_1_1.yaml:67]: Agreed, but how can that be achieved with the yaml syntax ?

I could not find an example that does this kind of negative check in other 1.1 tests.
for example:
https://github.com/project-chip/connectedhomeip/blob/master/src/app/tests/suites/certification/Test_TC_OO_1_1.yaml
https://github.com/project-chip/connectedhomeip/blob/master/src/app/tests/suites/certification/Test_TC_LVL_1_1.yaml

Maybe it's easier done in a python based test like: https://github.com/project-chip/connectedhomeip/blob/master/src/python_testing/TC_DeviceBasicComposition.py in one big sweep for all clusters on the device ?

Comment (bzbarsky-apple) [reply src/app/tests/suites/certification/Test_TC_TSUIC_1_1.yaml:67]: > Agreed, but how can that be achieved with the yaml syntax ?

```
      PICS: "!TSUIC.S.A0002"
      command: "readAttribute"
      attribute: "AttributeList"
      response:
          constraints:
              type: list
              excludes: [2]
```

Comment (bzbarsky-apple) [reply src/app/tests/suites/certification/Test_TC_TSUIC_1_1.yaml:67]: And there is a bunch of it in the newer 1_1 tests: 
```
% git grep -l excludes | grep yaml | grep 1_1
src/app/tests/suites/certification/Test_TC_CDOCONC_1_1.yaml
src/app/tests/suites/certification/Test_TC_CMOCONC_1_1.yaml
src/app/tests/suites/certification/Test_TC_FLDCONC_1_1.yaml
src/app/tests/suites/certification/Test_TC_NDOCONC_1_1.yaml
src/app/tests/suites/certification/Test_TC_OZCONC_1_1.yaml
src/app/tests/suites/certification/Test_TC_PMHCONC_1_1.yaml
src/app/tests/suites/certification/Test_TC_PMICONC_1_1.yaml
src/app/tests/suites/certification/Test_TC_PMKCONC_1_1.yaml
src/app/tests/suites/certification/Test_TC_RNCONC_1_1.yaml
src/app/tests/suites/certification/Test_TC_TIMESYNC_1_1.yaml
src/app/tests/suites/certification/Test_TC_TVOCCONC_1_1.yaml
```

Comment (tima-q) [reply src/app/tests/suites/certification/Test_TC_TSUIC_1_1.yaml:78]: @bzbarsky-apple Thanks for the reference - updated TC with the exclusion part.
Can you re-check ?

**Review by bzbarsky-apple — APPROVED** [review]

**Comment by tima-q** [comment]
Closing PR - had to be merged at the time - now already way to stale.

**Pull request closed without merging**